ci/webhook-receiver.py

```diff
@@ -12,6 +15,30 @@
 
 app = Flask(__name__)
 
+def verify_request_ip(request):
+    src_ip = ip_address(u'{}'.format(request.access_route[0]))
+    valid_ips = requests.get('https://api.github.com/meta').json()['hooks']
+
+    for ip in valid_ips:
+        if src_ip in ip_network(ip):
+            return True
+
+    return False
+
+def verify_request_secret(request):
```

Comment:
we talked offline, but i would like dealing with the secret to be like this:
* secret encrypted when stored at rest on disk
* ci server loads secret and enc key
* ci server decrypts secret
* ci server calculates and stores hash of secret
* ci server frees secret, enc secret, and key (ensure mem is free and cleared)